exporters/stdout/stdoutmetric/exporter_test.go

```diff
@@ -178,3 +195,275 @@ func TestAggregationSelector(t *testing.T) {
 	var unknownKind metric.InstrumentKind
 	assert.Equal(t, metric.AggregationDrop{}, exp.Aggregation(unknownKind))
 }
+
+func TestExporter_Export_Observability(t *testing.T) {
+	componentNameAttr := semconv.OTelComponentName("go.opentelemetry.io/otel/exporters/stdout/stdoutmetric.exporter/0")
+	componentTypeAttr := semconv.OTelComponentTypeKey.String(
+		"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric.exporter",
+	)
+	wantErr := errors.New("encoding failed")
+
+	tests := []struct {
+		name                  string
+		ctx                   context.Context
+		exporterOpts          []stdoutmetric.Option
+		observabilityEnabled  bool
+		expectedExportedCount int64
+		inflightAttrs         attribute.Set
+		attributes            attribute.Set
+		wantErr               error
+	}{
+		{
+			name:                  "Enabled",
+			ctx:                   t.Context(),
+			exporterOpts:          []stdoutmetric.Option{testEncoderOption()},
+			observabilityEnabled:  true,
+			expectedExportedCount: 19,
+			inflightAttrs:         attribute.NewSet(componentNameAttr, componentTypeAttr),
+			attributes:            attribute.NewSet(componentNameAttr, componentTypeAttr),
+		},
+		{
+			name:                  "Disabled",
+			ctx:                   t.Context(),
+			exporterOpts:          []stdoutmetric.Option{testEncoderOption()},
+			observabilityEnabled:  false,
+			expectedExportedCount: 0,
+		},
+		{
+			name:                  "EncodingError",
+			ctx:                   t.Context(),
+			exporterOpts:          []stdoutmetric.Option{stdoutmetric.WithEncoder(failingEncoder{})},
+			observabilityEnabled:  true,
+			expectedExportedCount: 19,
+			inflightAttrs:         attribute.NewSet(componentNameAttr, componentTypeAttr),
+			attributes: attribute.NewSet(
+				componentNameAttr,
+				componentTypeAttr,
+				semconv.ErrorType(wantErr),
+			),
+			wantErr: wantErr,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Setenv("OTEL_GO_X_OBSERVABILITY", strconv.FormatBool(tt.observabilityEnabled))
+
+			// Reset the exporter ID counter to ensure consistent component names
+			_ = counter.SetExporterID(0)
+
+			reader := metric.NewManualReader()
+			mp := metric.NewMeterProvider(metric.WithReader(reader))
+			origMp := otel.GetMeterProvider()
+			otel.SetMeterProvider(mp)
+			t.Cleanup(func() { otel.SetMeterProvider(origMp) })
+
+			exp, err := stdoutmetric.New(tt.exporterOpts...)
+			require.NoError(t, err)
+			rm := &metricdata.ResourceMetrics{ScopeMetrics: scopeMetrics()}
+
+			err = exp.Export(tt.ctx, rm)
+			if tt.wantErr != nil {
+				assert.EqualError(t, err, tt.wantErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			var metrics metricdata.ResourceMetrics
+			err = reader.Collect(tt.ctx, &metrics)
+			require.NoError(t, err)
+
+			if !tt.observabilityEnabled {
+				assert.Empty(t, metrics.ScopeMetrics)
+			} else {
```

Comment:
The `else` indentation is not needed.
```suggestion
			if !tt.observabilityEnabled {
				assert.Empty(t, metrics.ScopeMetrics)
				continue
			}
```